R/estimateDiversity.R

```diff
@@ -239,17 +294,70 @@ setMethod("estimateDiversity", signature = c(x = "SummarizedExperiment"),
     vegan::fisher.alpha(t(mat))
 }
 
+.calc_pd <- function(mat, tree, ...){
+
+    # Gets name of the taxa
+    taxa <- rownames(mat)
+    # Gets name of the samples
+    samples <- colnames(mat)
+
+    # Loops through every sample
+    PDs <- BiocParallel::bplapply(samples, function(sample){
+
+        # Stores taxa that are present in the sample, i.e., abundance is over 0
+        present <- taxa[ mat[, sample] > 0 ]
```

Comment:
I think this can be precalculated outside the `lapply`. `mat` and `samples` have the same length on the column and `mat` and `taxa` on the rows

Reply:
+1

Reply:
I have been trying to make this work with this approach. The problem is that, I want to get those taxa that are present in specific sample (e.g. vector that contains taxa for sample 1, taxa for sample 2...), and I haven't been able to do that. 

It is easy to get all the taxa that are present, but then it is just one big vector, so I lost the sample information (which taxa belongs to which sample). 

I can get list of samples, that include TRUE/FALSE if taxa is present, but then it is not possible to subset without using lapply, and then I do not know what is the benefit compared to the first approach. 

So, I'm little bit stuck with this

Reply:
Use `split` with the `lengths`/`sum` (aka. the number) of taxa found. Should yield a `CharacterList` and allow you to work with that

Reply:
Maybe also have a look at `PartitioningByEnd`/`PartitioningByWidth` and the rational behind that. `relist` is also related to that and `split`

Reply:
Ok, thanks for help

Reply:
Structure changed accordingly